plugins/glustershd/rest.go

```diff
@@ -187,3 +190,280 @@ func glustershDisableHandler(w http.ResponseWriter, r *http.Request) {
 
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, nil)
 }
+
+func glustershInfoHandler(w http.ResponseWriter, r *http.Request) {
+	p := mux.Vars(r)
+	volname := p["name"]
+	option := p["opts"]
+	_ = option
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	//validate volume name
```

Comment:
All these validations should be in a step function and be done under a lock.

Reply:
Move this and checks below into a step function. You will lose the ability to return `http.StatusNotFound` (404) but that is okay. The stepfunction can return error if volume not found.